web_poet/serialization/api.py

```diff
@@ -69,31 +104,49 @@ def register_serialization(
     f_serialize.f_deserialize = f_deserialize
 
 
-def deserialize_leaf(t: Type[T], data: SerializedLeafData) -> T:
-    f_ser: SerializeFunction = serialize_leaf.dispatch(t)
-    return f_ser.f_deserialize(t, data)
+def deserialize_leaf(cls: Type[T], data: SerializedLeafData) -> T:
+    f_ser: SerializeFunction = serialize_leaf.dispatch(cls)
+    return f_ser.f_deserialize(cls, data)
+
 
+def _get_fqname(cls: type) -> str:
+    """Return the fully qualified name for a type.
 
-def _get_fqname(t: type) -> str:
-    return f"{t.__module__}.{t.__qualname__}"
+    >>> _get_fqname(Injectable)
+    'web_poet.pages.Injectable'
+    """
+    return f"{cls.__module__}.{cls.__qualname__}"
 
 
 def serialize(deps: List[Any]) -> SerializedData:
-    # we skip injectable classes though the interface should probably not accept them at all
-    return {
-        _get_fqname(dep.__class__): serialize_leaf(dep)
-        for dep in deps
-        if not is_injectable(dep.__class__)
-    }
+    result: SerializedData = {}
+    for dep in deps:
+        cls = dep.__class__
+        if is_injectable(cls):
+            raise ValueError(f"Injectable type {cls} passed to serialize()")
+        result[_get_fqname(cls)] = serialize_leaf(dep)
+    return result
 
 
 def _load_type(type_name: str) -> Optional[type]:
```

Comment:
What do you think about renaming it to `_load_type_or_none`?
Also, an option is to just make it raise an exception; no strong opinion on that.